stubs/openpyxl/openpyxl/packaging/relationship.pyi

```diff
@@ -37,5 +40,12 @@ class RelationshipList(Serialisable):
     def to_tree(self): ...
 
 def get_rels_path(path): ...
-def get_dependents(archive, filename): ...
-def get_rel(archive, deps, id: Incomplete | None = None, cls: Incomplete | None = None): ...
+def get_dependents(archive: ZipFile, filename: str) -> RelationshipList: ...
+@overload
+def get_rel(
+    archive: ZipFile, deps: RelationshipList, id: str, cls: type[_SerialisableT]
+) -> _SerialisableT: ...  # incomplete: this could be restricted further from "Serialisable"
+@overload
+def get_rel(
+    archive: ZipFile, deps: RelationshipList, id: str | None = None, *, cls: type[_SerialisableT]
+) -> _SerialisableT: ...  # incomplete: this could be restricted further from "Serialisable"
```

Comment:
`cls` is always used at runtime: https://foss.heptapod.net/openpyxl/openpyxl/-/blob/branch/3.1/openpyxl/packaging/relationship.py#L143-150

About the `# incomplete`: Only `CacheDefinition`, `RecordList` and `TableDefinition` seem to define `rel_type`. (they also define is as ClassVars, which matches expected behaviour). But I can't find *any* class defining `deps`. The attribute seems to only be assigned in `get_rel` and used in `WorkbookParser.pivot_caches`. So I'm not exactly sure about which `Serialisable` child I should restrict to, if any.

Reply:
The more I look into it, the more I realize `obj.deps` is for internal used only and short-lived. Udated commit should be accurate.